### DISCUSSION TITLE
How to test Select widget?

### DISCUSSION THREAD

**Discussion by UlrichGoebel** [Textualize/textual#4514] [Q&A]
I try to test a self written bib which implement a few special widgets.

I use `pytest` and to test a `Select` Widget:

```python

from textual.app import App
from textual.widgets import Select

class Field():
    """"Comes from my bib and will be something more complex, of course"""
    def __init__(self):
        self.widget = Select((('One', 1), ('Two', 2), ('Three')))

class Main(App):
    def compose(self):
        field = Field()
        yield field.widget

class Test:
    app = Main()
    async with app.run_test() as pilot:
        # to make visible the options:
        await pilot.click(Select)
        # here I would like to click on one of these options
        await pilot.click(...)
```

So what could I put in the second `click` to chose the first, second or third option?

Onother way to test the `Select` coult be to simulate keystrokes instead of mouse clicks. Is there a possibility to simulate keys other then characters (like `pilot.press('a')`), let's say `Space`, `Down`, `Up` `Enter`, or even `Ctrl-Up` and so on? In the docu for [Pilot.press()](https://textual.textualize.io/api/pilot/#textual.pilot.Pilot.press) I just found normal `str` as argument for `pilot.press()`.

Thank You for any hint.

**Comment by willmcgugan** [comment]
You can use the name of the key e.g. `pilot.press("space")`

Run the `textual keys` command to see the key names.

**Reply by UlrichGoebel** [reply]
That's great.
